File: lib/features/video/services/spider/spider_adapter.dart
import 'spider_models.dart';

/// 蜘蛛适配器抽象接口
abstract interface class SpiderAdapter {
  /// 唯一标识
  String get key;

  /// 显示名称
  String get name;

  /// 蜘蛛类型
  SpiderType get type;

  /// 是否支持搜索，默认 true
  bool get isSearchSupported => true;

  /// 初始化
  Future<void> init(Map<String, dynamic> config);

  /// 首页内容
  Future<SpiderHomeResult> homeContent({int page = 1});

  /// 分类内容
  Future<SpiderListResult> categoryContent({
    required String tid,
    int page = 1,
    Map<String, String>? filter,
  });

  /// 详情内容
  Future<SpiderDetailResult> detailContent({required String id});

  /// 搜索内容
  Future<SpiderListResult> searchContent({
    required String keyword,
    int page = 1,
  });

  /// 播放内容
  Future<SpiderPlayResult> playerContent({
    required String flag,
    required String id,
  });

  /// 释放资源，默认空实现
  void dispose() {}
}

File: _temp_spider/SpiderAdapter.kt
package com.mediamix.shared.spider

import com.mediamix.shared.models.*

/**
 * 蜘蛛适配器接口
 *
 * 所有蜘蛛（CMS / JSON / XPath / JavaBridge）实现此接口，
 * 由 SpiderRegistry 统一管理并按需调用。
 */
interface SpiderAdapter {
    /** 唯一标识 */
    val key: String

    /** 显示名称 */
    val name: String

    /** 蜘蛛类型 */
    val type: SpiderType

    /** 是否支持搜索，默认 true */
    val isSearchSupported: Boolean get() = true

    /** 初始化 */
    suspend fun init(config: Map<String, Any>)

    /** 首页内容 */
    suspend fun homeContent(page: Int = 1): SpiderHomeResult

    /** 分类内容 */
    suspend fun categoryContent(
        tid: String,
        page: Int = 1,
        filter: Map<String, String>? = null,
    ): SpiderListResult

    /** 详情内容 */
    suspend fun detailContent(id: String): SpiderDetailResult

    /** 搜索内容 */
    suspend fun searchContent(keyword: String, page: Int = 1): SpiderListResult

    /** 播放内容 */
    suspend fun playerContent(flag: String, id: String): SpiderPlayResult

    /** 释放资源，默认空实现 */
    fun dispose() {}
}
